sdk/src/main/java/com/mapbox/maps/NativeObserver.kt

```diff
@@ -265,6 +265,15 @@ internal class NativeObserver(
     }
   }
 
+  // This method is needed to prevent receiving styleLoaded events with the old style
```

Comment:
@alexshalamov ptal, added resubscription to the STYLE_LOADED event that is called on every style change

@mapbox/maps-android is the comment clear at all?) Dirty hack but fixes the possible corner-case issues with loading multiple styles consecutively

Reply:
@yunikkk I was thinking whether we could push this logic down to the core renderer. Maybe for some of the events, gl-native could add `cancel and re-subscribe` code when new style is loaded, so that both iOS and Android platform would benefit from the fix.

Meanwhile, workaround on platform side should be sufficient for corner-cases.